rig/machine_control/boot.py

```diff
@@ -0,0 +1,160 @@
+"""Boot constructs for a SpiNNaker machine.
+
+Implementation is reconstructed from a Perl implementation which forms a
+significant part of the documentation for this process.
+"""
+from . import struct_file
+import socket
+import struct
+import time
+
+# Specifies the size of packets that should be sent to SpiNNaker to boot the
+# board, and to which port.
+BOOT_MAX_BLOCKS = 32  # 32 kiB limit in DTCM
+BOOT_WORD_SIZE = 256
+BOOT_BYTE_SIZE = 1024
+PORT = 54321
+BOOT_DELAY = 0.01  # Pause between transmitting boot packets
+
+# Specifies where in the boot data the struct containing default values should
+# be written, and how many bytes of it should be written.
+BOOT_DATA_OFFSET = 3*128
+BOOT_DATA_LENGTH = 128
+
+spin1_boot_options = {
+    "width": 2, "height": 2, "hardware_version": 0,
+    "led_config": 0x00076104,
+}
+spin2_boot_options = {
+    "width": 2, "height": 2, "hardware_version": 2,
+    "led_config": 0x00006103,
+}
+spin3_boot_options = {
+    "width": 2, "height": 2, "hardware_version": 3,
+    "led_config": 0x00000502,
+}
+spin4_boot_options = {
+    "width": 8, "height": 8, "hardware_version": 4,
+    "led_config": 0x00000001,
+}
+spin5_boot_options = {
+    "width": 8, "height": 8, "hardware_version": 5,
+    "led_config": 0x00000001,
+}
+
+
+def boot(hostname, width, height, cpu_frequency=200, hardware_version=0,
+         led_config=0x00000001, boot_data=None, struct_data=None):
+    """Boot a SpiNNaker machine of the given size.
+
+    Parameters
+    ----------
+    hostname : str
+        Hostname or IP address of the SpiNNaker board to boot.
+    width : int
+        Width of the machine (0 < w < 256)
+    height : int
+        Height of the machine (0 < h < 256)
+    cpu_frequency : int
+        CPU clock-frequency.  **Note**: The default (200 MHz) is known
+        safe.
+    hardware_version : int
+    led_config : int
+    boot_data : bytes
+        Data to boot the machine with.
+    struct_data : bytes or None
+        Data to interpret as a representation of the memory layout of data
+        within the boot data.
+
+    Notes
+    -----
+    The constants `rig.boot.spinX_boot_options` can be used to specify boot
```

Comment:
`rig.machine_control.boot.spinX_boot_options`

That said, I would prefer such constants lived in that `boot` module...